Description: Remove bundled libclang + fix for latest clang
 Use libclang as specified by llvm-config, and remove usages of
 CXTranslationUnit_CXX* flags which no longer exist.
Author: James Cowgill <james410@cowgill.org.uk>
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -366,9 +366,10 @@ elseif (MINGW)
     endif ()
   endif ()
 elseif (UNIX)
-  set( CLANG_INCLUDE "${CL_SRC_ROOT}/sdk/clang/Linux/${ARCH_NAME}/include" )
-  set( CLANG_LIBRARY "-L${CL_SRC_ROOT}/sdk/clang/Linux/${ARCH_NAME}/lib -lclang" )
-  set( CLANG_BINARY "${CL_SRC_ROOT}/sdk/clang/Linux/${ARCH_NAME}/lib/libclang.so")
+  execute_process(COMMAND llvm-config --includedir OUTPUT_VARIABLE CLANG_INCLUDE OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND llvm-config --libdir OUTPUT_VARIABLE CLANG_LIBDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+  set( CLANG_LIBRARY "-L${CLANG_LIBDIR} -lclang" )
+  set( CLANG_BINARY "" )
 endif()
 
 message( "-- clang link line ${CLANG_LIBRARY} ")
--- a/LiteEditor/clang_pch_maker_thread.cpp
+++ b/LiteEditor/clang_pch_maker_thread.cpp
@@ -434,12 +434,10 @@ CXTranslationUnit ClangWorkerThread::DoC
     // First time, need to create it
     unsigned flags;
     if(reparse) {
-        flags = CXTranslationUnit_CXXPrecompiledPreamble
-                | CXTranslationUnit_CacheCompletionResults
+        flags = CXTranslationUnit_CacheCompletionResults
                 | CXTranslationUnit_PrecompiledPreamble
                 | CXTranslationUnit_Incomplete
-                | CXTranslationUnit_DetailedPreprocessingRecord
-                | CXTranslationUnit_CXXChainedPCH;
+                | CXTranslationUnit_DetailedPreprocessingRecord;
     } else {
         flags = CXTranslationUnit_Incomplete
                 | CXTranslationUnit_SkipFunctionBodies
